saleor/graphql/reservation/tests/test_reservation_creation.py

```diff
@@ -0,0 +1,223 @@
+):
+    variant = variant_with_reserved_stock
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    variables = {
+        "reservationInput": {
+            "countryCode": "US",
+            "quantity": 2,
+            "variantId": variant_id,
+        }
+    }
+    reservation = Reservation.objects.first()
+    assert reservation.quantity == 3
+    response = user_api_client.post_graphql(MUTATION_RESERVATION_CREATE, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["reservationCreate"]
+    assert not data["reservationErrors"]
+
+    reservation.refresh_from_db()
+    assert reservation.quantity == 2
+
+
+def test_mutation_updates_expired_stock_reservation(
+    customer_user, user_api_client, variant_with_expired_stock_reservation
+):
+    variant = variant_with_expired_stock_reservation
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    variables = {
+        "reservationInput": {
+            "countryCode": "US",
+            "quantity": 2,
+            "variantId": variant_id,
+        }
+    }
+    expired_reservation = Reservation.objects.first()
+    assert expired_reservation
+    response = user_api_client.post_graphql(MUTATION_RESERVATION_CREATE, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["reservationCreate"]
+    assert not data["reservationErrors"]
+
+    reservation = Reservation.objects.first()
+    assert reservation == expired_reservation
+    assert reservation.expires > expired_reservation.expires
+
+
+def test_mutation_fails_if_user_is_not_authenticated(
+    api_client, variant_with_many_stocks
+):
+    variant = variant_with_many_stocks
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    variables = {
+        "reservationInput": {
+            "countryCode": "US",
+            "quantity": 2,
+            "variantId": variant_id,
+        }
+    }
+    assert not Reservation.objects.exists()
+    response = api_client.post_graphql(MUTATION_RESERVATION_CREATE, variables)
+    content = get_graphql_content(response, ignore_errors=True)
+    assert content["errors"][0]["message"] == (
+        "You do not have permission to perform this action"
+    )
+    assert not Reservation.objects.exists()
+
+
+def test_mutation_fails_when_reserved_quantity_is_less_than_one(
+    user_api_client, variant_with_many_stocks
+):
+    variant = variant_with_many_stocks
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    variables = {
+        "reservationInput": {
+            "countryCode": "US",
+            "quantity": 0,
+            "variantId": variant_id,
+        }
+    }
+    assert not Reservation.objects.exists()
+    response = user_api_client.post_graphql(MUTATION_RESERVATION_CREATE, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["reservationCreate"]
+    assert data["reservationErrors"][0]["message"] == (
```

Comment:
In tests, we should compare `code` and `field` instead of the message.  It's should be fixed in all tests in this PR.